mirrord/agent/src/steal/http_traffic/hyper_handler.rs

```diff
@@ -39,6 +40,39 @@ where
     tokio::spawn(async move { tx.send(value).map_err(HttpTrafficError::from).await });
 }
 
+#[tracing::instrument(level = "debug", skip(tx))]
+fn intercepted_request(
+    request: Request<Incoming>,
+    tx: Sender<UnmatchedResponse>,
+    original_destination: SocketAddr,
+) {
+    tokio::spawn(async move {
+        let target_stream = TcpStream::connect(original_destination).await.unwrap();
+
+        let (mut request_sender, connection) =
+            client::conn::http1::handshake(target_stream).await.unwrap();
+
+        tokio::spawn(async move {
+            if let Err(fail) = connection.await {
+                eprintln!("Error in connection: {}", fail);
+            }
+        });
+
+        // TODO(alex) [mid] 2022-12-21: Does the host come in with the address of our intercepted
+        // stream, or of the original destination already?
+        //
+        // Doing this only because the local tests are not handled by the stealer mechanism, so the
+        // host comes with the wrong address.
+        // let proper_host = HeaderValue::from_str(&original_destination.to_string()).unwrap();
+        // request.headers_mut().insert(HOST, proper_host).unwrap();
+        let intercepted_response = request_sender.send_request(request).await.unwrap();
+
+        let response = UnmatchedResponse(intercepted_response);
+        // TODO(alex) [high] 2022-12-20: Send this response back in the original stream.
+        tx.send(response).map_err(HttpTrafficError::from).await
```

Comment:
Doesn't the `await` have to come before the `map_err`?
```suggestion
        tx.send(response).await.map_err(HttpTrafficError::from)
```

Why does it compile this way?

Reply:
Actually, why map that error at all, if nothing happens with the result?